### PR TITLE
Add keyword spotting demos

### PR DESCRIPTION
**Pull request by pblouw**
Included here are a notebook illustrating the process of training up a keyword spotter and deploying it on Loihi with an existing dataset, and a GUI script for running a live keyword spotting application using input from a microphone in core Nengo. This quality of this latter application depends quite a bit on the audio recording setup, so the idea was to release an example for people to play around absent any guarantees.

Currently, one CI test is failing on this PR due to a tornado `AttributeError` that I can't replicate locally. Suggestions on solving this (and on the PR more generally) are welcome!

### CODE REVIEW

**Comment by arvoelke** [comment]
The tornado issue might be related to https://github.com/nengo/nengo-loihi/commit/1517915baa1b0200e5241ba55a65e9d0bc09bff8 where we had to limit the version of `nbconvert`.

**Comment by pblouw** [comment]
Looking through the CI build log, I think you're right, though it appears that the `nbconvert` import is happening in Nengo. So perhaps the fix is to update the `.travis.yml` to do a developer Nengo install? (I think there have been a bunch of updates to the Travis CI stuff since the last Nengo release).

**Comment by pblouw** [comment]
I fixed this by downgrading tornado in the `.travis.yaml` file. Looks like this is a known issue involving the new release of tornado breaking jupyter in conda installs: https://github.com/jupyter/notebook/issues/4311

Tagging @drasmuss as I understand you are working on standardizing some of the Travis CI stuff across Nengo libraries.

**Comment by tbekolay** [comment]
This example no longer works properly on recent versions of TensorFlow. Even when using as much as possible from the `compat.v1` module and disabling eager execution, the loss explodes to a super large negative value. We also now have [this Loihi KWS example](https://www.nengo.ai/nengo-loihi/examples/keyword-spotting.html), so I'm going to close this @pblouw. But feel free to reopen if there's an easy fix that I'm missing!
